group-coordinator/src/test/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroupMemberTest.java

```diff
@@ -304,4 +312,75 @@ public void testUpdateWithConsumerGroupCurrentMemberAssignmentValue() {
+            .setAssignedPartitions(Collections.singletonList(new ConsumerGroupCurrentMemberAssignmentValue.TopicPartitions()
+                .setTopicId(topicId1)
+                .setPartitions(assignedPartitions)))
+            .setPartitionsPendingRevocation(Collections.singletonList(new ConsumerGroupCurrentMemberAssignmentValue.TopicPartitions()
+                .setTopicId(topicId2)
+                .setPartitions(Arrays.asList(3, 4, 5))))
+            .setPartitionsPendingAssignment(Collections.singletonList(new ConsumerGroupCurrentMemberAssignmentValue.TopicPartitions()
+                .setTopicId(topicId3)
+                .setPartitions(Arrays.asList(6, 7, 8))));
+        String memberId = Uuid.randomUuid().toString();
+        String clientId = "clientId";
+        String instanceId = "instanceId";
+        String rackId = "rackId";
+        String clientHost = "clientHost";
+        List<String> subscribedTopicNames = Arrays.asList("topic1", "topic2");
+        String subscribedTopicRegex = "topic.*";
+        Map<Uuid, Set<Integer>> assignmentMap = new HashMap<>();
+        assignmentMap.put(Uuid.randomUuid(), new HashSet<>());
+        Assignment targetAssignment = new Assignment(assignmentMap);
+        ConsumerGroupMember member = new ConsumerGroupMember.Builder(memberId)
+            .updateWith(record)
+            .setClientId(clientId)
+            .setInstanceId(instanceId)
+            .setRackId(rackId)
+            .setClientHost(clientHost)
+            .setSubscribedTopicNames(subscribedTopicNames)
+            .setSubscribedTopicRegex(subscribedTopicRegex)
+            .build();
+
+        ConsumerGroupDescribeResponseData.Member consumerGroupDescribeMember = member.asConsumerGroupDescribeMember(targetAssignment);
+
+        assertEquals(memberId, consumerGroupDescribeMember.memberId().toString());
+        assertEquals(clientId, consumerGroupDescribeMember.clientId());
+        assertEquals(instanceId, consumerGroupDescribeMember.instanceId());
+        assertEquals(rackId, consumerGroupDescribeMember.rackId());
+        assertEquals(clientHost, consumerGroupDescribeMember.clientHost());
+        assertEquals(subscribedTopicNames, consumerGroupDescribeMember.subscribedTopicNames());
+        assertEquals(subscribedTopicRegex, consumerGroupDescribeMember.subscribedTopicRegex());
+        assertEquals(
+            new ConsumerGroupDescribeResponseData.Assignment()
+                .setTopicPartitions(targetAssignment.partitions().entrySet().stream().map(
+                    item -> new ConsumerGroupDescribeResponseData.TopicPartitions()
+                        .setTopicId(item.getKey())
+                        .setPartitions(new ArrayList<>(item.getValue()))
+                ).collect(Collectors.toList())),
+            consumerGroupDescribeMember.targetAssignment()
+        );
+
+        assertEquals(assignedPartitions, consumerGroupDescribeMember.assignment().topicPartitions().get(0).partitions());
```

Comment:
nit: We usually prefer to construct the full expected data structure, here `ConsumerGroupDescribeResponseData.Member`, and to use `assertEquals` because it ensures that the all the fields are tested (also future fields).